app/models/designation.js

```diff
@@ -61,6 +62,58 @@ export default Model.extend({
     return this.get("state").capitalize();
   }),
 
+  stateIcon: Ember.computed('stateIcon', function () {
+    const state = this.get("state");
+    switch (state) {
+      case "awaiting_dispatch":
+      case "scheduled":
+        return "clock-o";
+      case "processing":
+        return "list";
+      case "submitted":
+        return "envelope";
+      case "dispatching":
+        return "paper-plane";
+      case "cancelled":
+        return "thumbs-down";
+      case "closed":
+        return "lock";
+      default:
+        return "";
+    }
+  }),
+
+  transportIcon: Ember.computed("transportIcon", function() {
+    const key = this.get("transportKey");
+    switch (key) {
+      case "gogovan_transport":
+        return "truck";
+      case "collection_transport":
+        return "male";
+      default:
+        return "";
+    }
+  }),
+
+  transportLabel: Ember.computed("transportLabel", function() {
+    const key = this.get('transportKey');
+    return this.get("i18n").t(`order_transports.${key}`);
+  }),
+
+  transportKey: Ember.computed("transportKey", function() {
```

Comment:
@patrixr , you can make this computed property dependent like this
`transportKey: Ember.computed("orderTransport.transportType", function() {`

Reply:
@Darshan-Chauhan I see, I mis-used the ember computed properties. I'll fix those :)